src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -498,6 +535,7 @@ void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::serialize(
     Archive& ar, const unsigned int /* version */)
 {
   ar & BOOST_SERIALIZATION_NVP(parameter);
+  ar & BOOST_SERIALIZATION_NVP(size);
```

Comment:
Probably we should do some special handling here since `size` is a newly serialized parameter, like this... uncomment `version`, then do

```
if (version > 1)
  ar & BOOST_SERIALIZATION_NVP(size);
else if (Archive::is_loading::value)
  size = 0; // Set size to a default value that will be inferred later.
```

then below the class definition, change the line `BOOST_STATIC_CONSTANT(int, value = 1)` to `BOOST_STATIC_CONSTANT(int, value = 2)`.

This will allow older saved models to be loaded even though they were saved without the `size` parameter.  (I think that 0 is a decent default value; maybe if there is a better way to figure it out you can do that.)

Reply:
Nice, looks better now, but I think we can remove the line this commented is connected to, otherwise `size` will be doubly serialized:

```
ar & BOOST_SERIALIZATION_NVP(size);
```

I can do that during merge if you like.